src/autora/experimentalist/inequality/__init__.py

```diff
@@ -58,89 +59,98 @@ def sample(
-        >>> summed_inequality_sampler([1, 1, 1, 2, 2, 2, 3, 3, 3], [1, 1, 2, 2, 2, 2, 3, 3, 3], 3)
+        >>> summed_inequality_sample([1, 1, 1, 2, 2, 2, 3, 3, 3], [1, 1, 2, 2, 2, 2, 3, 3, 3], 3)
         array([[1],
                [3],
                [1]])
 
         The experimentalist samples without replacemnt!
-        >>> summed_inequality_sampler([1, 2, 3], [1, 1, 1], 3)
+        >>> summed_inequality_sample([1, 2, 3], [1, 1, 1], 3)
         array([[3],
                [2],
                [1]])
 
     """
 
-    if isinstance(condition_pool, Iterable):
-        condition_pool = np.array(list(condition_pool))
+    X = np.array(conditions)
 
-    if isinstance(reference_conditions, Iterable):
-        reference_conditions = np.array(list(reference_conditions))
+    _reference_conditions = reference_conditions.copy()
+    if isinstance(reference_conditions, pd.DataFrame):
+        if set(conditions.columns) != set(reference_conditions.columns):
+            raise Exception(
+                f"Variable names {set(conditions.columns)} in conditions"
+                f"and {set(reference_conditions.columns)} in allowed values don't match. "
+            )
 
-    if condition_pool.ndim == 1:
-        condition_pool = condition_pool.reshape(-1, 1)
+        _reference_conditions = _reference_conditions[conditions.columns]
 
-    if reference_conditions.ndim == 1:
-        reference_conditions = reference_conditions.reshape(-1, 1)
+    X_reference_conditions = np.array(_reference_conditions)
 
-    if condition_pool.shape[1] != reference_conditions.shape[1]:
+    if X.ndim == 1:
+        X = X.reshape(-1, 1)
+
+    if X_reference_conditions.ndim == 1:
+        X_reference_conditions = X_reference_conditions.reshape(-1, 1)
+
+    if X.shape[1] != X_reference_conditions.shape[1]:
         raise ValueError(
-            f"condition_pool and reference_conditions must have the same number of columns.\n"
-            f"condition_pool has {condition_pool.shape[1]} columns, "
-            f"while reference_conditions has {reference_conditions.shape[1]} columns."
+            f"conditions and reference_conditions must have the same number of columns.\n"
+            f"conditions has {X.shape[1]} columns, "
+            f"while reference_conditions has {X_reference_conditions.shape[1]} columns."
         )
 
-    if condition_pool.shape[0] < num_samples:
+    if X.shape[0] < num_samples:
         raise ValueError(
-            f"condition_pool must have at least {num_samples} rows matching the number "
+            f"conditions must have at least {num_samples} rows matching the number "
             f"of requested samples."
         )
 
     dist = DistanceMetric.get_metric(metric)
 
-    # create a list to store the n condition_pool values with the highest inequality scores
+    # create a list to store the n conditions values with the highest inequality scores
     condition_pool_res = []
```

Comment:
should this be renamed to `conditions_res` and reflected in the rest of this file?